In [ ]:
from google.colab import files
files.upload()

Saving sample_input.json to sample_input.json


{'sample_input.json': b'{"sensor_1_roll_mean_6": 518.67, "sensor_1_roll_std_6": 0.0, "sensor_1_roll_mean_12": 518.67, "sensor_1_roll_std_12": 0.0, "sensor_2_roll_mean_6": 642.5, "sensor_2_roll_std_6": 0.3829882504725299, "sensor_2_roll_mean_12": 642.4333333333333, "sensor_2_roll_std_12": 0.3710754231919903, "sensor_3_roll_mean_6": 1586.0266666666666, "sensor_3_roll_std_6": 4.130998265149241, "sensor_3_roll_mean_12": 1586.9808333333333, "sensor_3_roll_std_12": 3.1430079927394297, "sensor_4_roll_mean_6": 1399.9516666666666, "sensor_4_roll_std_6": 4.4160861253675145, "sensor_4_roll_mean_12": 1401.7283333333332, "sensor_4_roll_std_12": 4.5729736363546625, "sensor_5_roll_mean_6": 14.62, "sensor_5_roll_std_6": 0.0, "sensor_5_roll_mean_12": 14.62, "sensor_5_roll_std_12": 0.0, "sensor_6_roll_mean_6": 21.61, "sensor_6_roll_std_6": 0.0, "sensor_6_roll_mean_12": 21.61, "sensor_6_roll_std_12": 0.0, "sensor_7_roll_mean_6": 553.8383333333334, "sensor_7_roll_std_6": 0.31205234603602805, "sensor_7_rol

In [ ]:
import json
import requests

with open("sample_input.json", "r") as f: #Reading data from a JSON file & Converts it into a Python dictionary
    sample_input = json.load(f)

url = "https://dispensational-doily-jonelle.ngrok-free.dev/predict" #This is the public API endpoint, Provided by ngrok

response = requests.post(url, json=sample_input) #Sending POST request

print(response.status_code)
print(response.json()) #Reading API response

200
{'failure_probability': 0.0, 'latency_ms': 8.19}


**Deployment Results**

The deployed Flask API successfully returns failure probabilities in real time. For a sample sensor input, the model predicted a failure probability of 0.0, indicating normal operating conditions. The average response latency was 8.19 ms, satisfying the real-time constraint (<50 ms).